rest_models/backend/compiler.py

```diff
@@ -1007,28 +1031,30 @@ def execute_sql(self, return_id=False, chunk_size=None):
         else:
             result_json = None
             for obj in query_objs:
-                data = {
-                    f.column: f.get_db_prep_save(
-                        getattr(obj, f.attname) if query.raw else f.pre_save(obj, True),
-                        connection=self.connection
-                    )
-                    for f in query.fields
-                    }
+                files = build_files(obj)
 
                 json = {
-                    get_resource_name(query.model, many=False): data
+                    get_resource_name(query.model, many=False): build_data(obj)
                 }
                 response = self.connection.cursor().post(
                     get_resource_path(self.query.model),
-                    json=json
+                    json=json,
+                    files=files
                 )
                 if response.status_code != 201:
-                    raise FakeDatabaseDbAPI2.ProgrammingError("error while creating %s with %s.\n%s" % (
-                        obj, json, message_from_response(response)))
+                    raise FakeDatabaseDbAPI2.ProgrammingError("error while creating %s with json=%s,files=%s.\n%s" % (
+                        obj, json, files, message_from_response(response)))
                 result_json = response.json()
                 new = result_json[get_resource_name(query.model, many=False)]
                 for field in opts.concrete_fields:
-                    setattr(obj, field.attname, field.to_python(new[field.concrete and field.db_column or field.name]))
+                    raw_val = new[field.concrete and field.db_column or field.name]
+                    if isinstance(field, FileField) and hasattr(field.storage, 'prepare_result_from_api'):
+                        python_val = field.storage.prepare_result_from_api(raw_val, self.connection)
```

Comment:
handle result of api like a normal get would do: store the full url for later, and just return the file name